osbuild/objectstore.py

```diff
@@ -348,68 +366,57 @@ def new(self, object_id: str):
         `Object.write`), these must manually be committed to the
         store via `commit()`.
         """
+        if not self._stack:
+            raise AssertionError("Need active context")
 
-        obj = Object(self, object_id, Object.Mode.WRITE)
+        obj = Object(self.cache, object_id, Object.Mode.WRITE)
+        self._stack.enter_context(obj)
 
         self._objs.add(obj)
 
         return obj
 
-    def commit(self, obj: Object, object_id: str) -> str:
+    def commit(self, obj: Object, object_id: str):
         """Commits a Object to the object store
 
-        Move the contents of the obj (Object) to object directory
-        of the store with a universally unique name. Creates a
-        symlink to that ('objects/{hash}') in the references
-        directory with the object_id as the name ('refs/{object_id}).
-        If the link already exists, it will be atomically replaced.
-
-        If object_id is different from the id of the object, a copy
-        of the object will be stored.
+        TBD
```

Comment:
`TBD` because we shortcut `commit()` to `clone()`?

Reply:
Ups, forgot to fill that in!